api/proto/disperser/disperser.proto

```diff
@@ -115,17 +127,39 @@ message BlobStatusReply {
 	BlobInfo info = 2;
 }
 
-// RetrieveBlobRequest contains parameters to retrieve the blob.
-message RetrieveBlobRequest {
+// GetBlobRequest contains parameters to retrieve the blob.
+message GetBlobRequest {
```

Comment:
I think we should allow two lookup methods, as specified in this proposal: https://www.notion.so/eigen-labs/Unified-blob-addressal-in-EigenDA-ae426c610f9e4e61b2e46eec76f7ca4d?d=f09f902f4d0743a7abef61faae0e2045#d392986bccbc43f9b6f8bcf6ace29255

1. BlobAuthHeaderHash
2. BatchHeaderHash + BlobIndex

Reply:
change made

Reply:
Are we planning on deprecating lookup via batch header hash? The answer to this question will determine the best way to add the secondary lookup strategy.

# If we will always support both lookup mechanisms

Make `GetBlobRequest` contain a `oneof` argument where the caller can either provide the auth header hash or the batch header hash + blob index.

# If we will eventually stop supporting lookup by batch header hash

Leave the old and deprecated "RetrieveChunkRequest" using the batch header hash, and the have the new "GetChunkRequest" use the auth header hash.

Reply:
Hmm, this is a good question which is difficult to immediately answer. I think there's a possibility that we would deprecate the old one, but probably not before we wanted to start also using the new one. Which option would you go with in that world?